api/v1alpha1/gateway_types.go

```diff
@@ -613,6 +630,10 @@ type GatewayStatus struct {
 	// otherwise it will use the service URL
 	// +optional
 	URL string `json:"url,omitempty"`
+
+	// ServiceAccountName is the name of the created service account
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```

Comment:
```suggestion
```

Not sure why do we need a ServiceAccount in the HPA spec. We should use the deployment spec only.